### ISSUE TITLE
Minishift cleanup steps

### ISSUE THREAD

**Issue by ALRubinger** [minishift/minishift#701]
1.0.0.rc.1 has some issues with our work in appdev.openshift.io.  We're needing to downgrade local envs to use Beta5.

```
[alr@tubescreamer minishift]$ rm -rf ~/.minishift/
[alr@tubescreamer minishift]$ minishift start --memory=4096
Starting local OpenShift cluster using 'kvm' hypervisor...
Downloading ISO 'https://github.com/minishift/minishift-b2d-iso/releases/download/v1.0.2/minishift-b2d.iso'
 40.00 MB / 40.00 MB [=================] 100.00% 0s
E0406 13:56:18.369253   27603 start.go:160] Error starting the VM: Error creating the VM. Error creating machine: Error in driver during machine creation: [Code-9] [Domain-20] operation failed: domain 'minishift' already exists with uuid 9016d6b8-dedc-465d-8d5b-9a9beb3aeb1e. Retrying.
E0406 13:59:24.599334   27603 start.go:160] Error starting the VM: Error configuring authorization on host: Too many retries waiting for SSH to be available.  Last error: Maximum number of retries (60) exceeded. Retrying.
```

Help?

**Comment by jorgemoralespou** [comment]
rm -rf is not a good solution, although I've seen it suggested many times, if you have the VM already created, so only should be run after __minishift delete__

I would recommend you deleting the VM from KVM, although I don't know how to do that.

Maybe @budhrg or @gbraad or @praveenkumar

**Comment by LalatenduMohanty** [comment]
@ALRubinger  as @jorgemoralespou mentioned the issue seems to be with your setup and `$ minishift delete` is the ideal way to do it. However if that does not apply and you want completely delete everything and start from fresh here are the steps

*  Check if the VM is still running `$ sudo virsh list --all`
* If it is running, stop the minishift VM i.e. `$ sudo virsh destroy minishift`
* Delete the VM `$ sudo virsh undefine minishift`
*  Delete the .minishift/machines dir i.e. $ rm -rf ~/.minishift/machines. Or as an extreme step `$rm -rf ~/.minishift`

**Comment by ALRubinger** [comment]
@jorgemoralespou @LalatenduMohanty You guys fixed me up!  If you want to point me to a docs place I can contribute that (or where it already exists) I'll be happy to pay it forward.

**Comment by jorgemoralespou** [comment]
That should be @LalatenduMohanty and possibly @thatdocslady

**Comment by LalatenduMohanty** [comment]
@ALRubinger  it should be the https://github.com/minishift/minishift/blob/master/docs/source/using/troubleshooting.adoc . Unfortunately we do not have the cleanup steps documented. So you are welcome to contribute.  

Please also refer https://github.com/minishift/minishift/blob/master/docs/source/developing/writing-docs.adoc before you contribute.

**Comment by hferentschik** [comment]
@LalatenduMohanty they are here - https://github.com/minishift/minishift/blob/master/docs/source/using/troubleshooting.adoc#error-starting-the-vm--operation-failed-domain-minishift-already-exists-

**Comment by LalatenduMohanty** [comment]
@hferentschik It has some commands, but do not have all the steps to completely cleanup an existing setup. Which I think should have been there.

**Comment by hferentschik** [comment]
So what's the action to be taken here? We could refine the trouble-shooting guide? I think this makes sense, especially looking at the hosted docs - https://docs.openshift.org/latest/minishift/using/troubleshooting.html

For a trouble shooting guide it is imo important to have all covered topics listed in the table of contents, but with the new docs that is not the case and we limit to first level only. For this page we should change this. While on it I could image a bit nicer formatting overall as well.

**Comment by LalatenduMohanty** [comment]
> We could refine the trouble-shooting guide? I think this makes sense, especially looking at the hosted docs - https://docs.openshift.org/latest/minishift/using/troubleshooting.html

+1. Lets do that.

**Comment by hferentschik** [comment]
@Preeticp, thanks for taking this. I think apart of making sure the covered issues are listed at the toc (just a variable change really), we should also look for a better way of separating the various topics. Looking at https://docs.openshift.org/latest/minishift/using/troubleshooting.html I have a hard time distinguishing where one troubleshooting point starts and where it ends. 

Also, I think we might want to get rid of grouping by driver, instead have single dedicated problem/solution paragraphs.

**Comment by Preeticp** [comment]
Sure I will do that.

**Comment by LalatenduMohanty** [comment]
@Preeticp I think we should have generic topic for cleanup then we can give the example for KVM. As the steps (not the commands) are applicable to all hypervisors in general.

**Comment by Preeticp** [comment]
As for generic information, it already exists in the Uninstalling section: https://github.com/minishift/minishift/blob/master/docs/source/getting-started/installing.adoc#uninstalling-minishift
I will update instructions for KVM in troubleshooting, make TOC level changes as per @hferentschik to make the errors visible and link the section to Uninstalling section.

**Comment by hferentschik** [comment]
Merged via pull request #717
